listener/avalon_listener/tcs_work_order_handler_sync.py

```diff
@@ -126,12 +126,17 @@ def WorkOrderSubmit(self, **params):
             )
         if((self.workorder_count + 1) > self.max_workorder_count):
 
+            # wo_ids is a csv of work order ids retrieved from database
+            wo_ids = self.kv_helper.get("wo-worker-processed", worker_id)
+            processed_wo_ids = [] if wo_ids is None else wo_ids.split(",")
+
             # if max count reached clear a processed entry
             work_orders = self.kv_helper.lookup("wo-timestamps")
             for id in work_orders:
                 # If work order is processed then remove from table
-                if(self.kv_helper.get("wo-processed", id) is not None):
-                    self.kv_helper.remove("wo-processed", id)
+                if id in processed_wo_ids:
+                    self.kv_helper.csv_search_delete(
+                        "wo-worker-processed", worker_id, id)
                     self.kv_helper.remove("wo-requests", id)
                     self.kv_helper.remove("wo-responses", id)
                     self.kv_helper.remove("wo-receipts", id)
```

Comment:
Lot of duplicate code in WorkOrderSubmit() function for sync and async handling. This could be cleaned up (separate PR)